File: robot/Comm/Messages/Remote/FromRobot/LogBeginMessage.cpp
#include <string.h>

#include "LogBeginMessage.h"

LogBeginMessage::LogBeginMessage(unsigned long timestamp)
  : RemoteMessageFromRobot(RemoteMessageFromRobot::logBegin, 0, NULL) {
  createData(timestamp);
}

LogBeginMessage::~LogBeginMessage() {
}

void LogBeginMessage::createData(unsigned long timestamp) {
  data = new char[4];
  if (data == NULL) {
    size = 0;
    return;
  }

  writeInt(data, timestamp);
  size = 4;
}

File: tools/SharedJavaLibrary/src/network/message/fromrobot/LogBeginMessage.java
package network.message.fromrobot;

import network.message.RemoteMessage;
import network.message.RemoteMessageFromRobot;

public class LogBeginMessage extends RemoteMessageFromRobot {
  
  public LogBeginMessage(int timestamp) {
    super(RemoteMessageFromRobot.MessageType.LOG_BEGIN, null);
    
    this.timestamp = timestamp;
  }
  
  public int getTimestamp() {
    return timestamp;
  }
  
  private int timestamp;

  public static LogBeginMessage create(byte [] data) {
    try {
      int timestamp = RemoteMessage.readInt(data, 0);
      
      return new LogBeginMessage(timestamp);
    }
    catch (Exception e) {
      return null;
    }
  }

}
